File: python/FactoryMethod/empty/Store.py
from Product import Product

class Store(object):

    def __init__(self):
        """
        init an empty list of products
        """

    def add(self, name):
        """
        create a new product with the product classe's
        make_product method and add it to our list
        """

    def remove(self, name):
        """
        find the product with 'name' using the product classe's
        is_equal method and delete it
        """

    def get_inventory(self):
        """
        returns a string of the current products - with a newline after each
        """

    def make_product(self, name):
        """
        'abstract' method - should be implemented in child classes- content
        here should reflect that
        """

File: java/FactoryMethod/empty/Store.java
import java.util.ArrayList;

abstract class Store{

  public ArrayList<Product> products = new ArrayList<Product>();

  /*
  create a new product with the product class'
  make_product method and add it to our list
  */
  public void add(String name){
  }

  /*
  find the product with 'name' in our Product ArrayList using the product class'
  is_equal  method and delete it
  */
  public void remove(String name){
  }

  /*
  return a string of the current products, with a newline after each
  */
  public String get_inventory(){
  }

  /*
  add the make_product(String name) method here
  an abstract method that creates a new product, this is the Factory method
  implemented in children classes
  */
}
